Assets/Plugins/Editor/Uplift/Windows/UpdateWindow.cs

```diff
@@ -35,165 +35,103 @@ namespace Uplift.Windows
-            string installed_version = installed ? upbring.GetInstalledPackage(name).Version : "";
-
-            if (installed)
+            if(installed != null)
             {
-                EditorGUILayout.LabelField("- Installed version is " + installed_version);
-            }
-            else
-            {
-                EditorGUILayout.LabelField("- Not yet installed");
-            }
+                EditorGUILayout.LabelField("Installed version: " + installed.Version);
 
-            if (!installable)
-            {
-                EditorGUILayout.HelpBox("No repository contains this package. Try specifying one whith this package in.", MessageType.Warning);
-            }
-            else
-            {
-                PackageRepo latestPackageRepo = packageList.GetLatestPackage(name);
-                string latest_version = latestPackageRepo.Package.PackageVersion;
-
-                EditorGUILayout.LabelField(string.Format("- Latest version is: {0} (from {1})", latest_version, latestPackageRepo.Repository.ToString()));
-                if (!definition.Requirement.IsMetBy(latest_version))
+                if(installed.Version != bestMatch.Package.PackageVersion)
                 {
-                    EditorGUILayout.HelpBox("The latest available version does not meet the requirement of the dependency.", MessageType.Warning);
+                    EditorGUILayout.HelpBox(
+                        string.Format(
+                            "Package is outdated. You can update it to {0} (from {1})",
+                            bestMatch.Package.PackageVersion,
+                            bestMatch.Repository.ToString()
+                        ),
+                        MessageType.Info
+                    );
+                    if(GUILayout.Button("Update to version " + bestMatch.Package.PackageVersion))
+                    {
+                        UpliftManager.Instance().UpdatePackage(bestMatch);
+                        Init();
+                        Repaint();
+                    }
                 }
+                else
+                    EditorGUILayout.HelpBox("Package is up to date!", MessageType.Info);
 
-                GUI.enabled = definition.Requirement.IsMetBy(latest_version) && installed && installed_version != latest_version;
-                if (GUILayout.Button("Update to " + latest_version))
-                {
-                    Debug.Log(string.Format("Updating package {0} (to {1})", name, latest_version));
-                    manager.UpdatePackage(latestPackageRepo);
-
-                    AssetDatabase.Refresh();
-
-                    Repaint();
-                }
-                GUI.enabled = true;
+                if(!definition.Requirement.IsMetBy(installed.Version))
+                    EditorGUILayout.HelpBox(
+                        "The version of the package currently installed does not match the requirements of your project!",
```

Comment:
maybe display the version and the requirements?

Reply:
I believe that this is not necessary as it is displayed just above:
![capture d ecran 9](https://user-images.githubusercontent.com/16774018/32499224-4d0e45b2-c3d2-11e7-82b0-c5dd28a816a6.png)

Reply:
Fine as is.